src/azul/service/manifest_service.py

```diff
@@ -2059,6 +2062,20 @@ def create_file(self) -> tuple[str, str | None]:
         replicas = list(self._list_replicas())
         plugin = self.metadata_plugin
         replica_schemas = plugin.verbatim_pfb_schema(replicas)
+        # FIXME: Move injection of snapshot ID field to metadata plugin
+        #        https://github.com/DataBiosphere/azul/issues/7411
+        if config.is_anvil_enabled(self.catalog):
+            for replica in replicas:
+                if replica['replica_type'] == 'anvil_dataset':
+                    source_id = replica['source']['id']
+                    replica['contents']['source_datarepo_snapshot_id'] = source_id
```

Comment:
```suggestion
                    replica['contents']['datarepo_snapshot_id'] = source_id
```

https://github.com/DataBiosphere/azul/issues/7254#issuecomment-3314624470